### ISSUE TITLE
With 4.1.5 I am getting in some cases Type errors on segmentation

### ISSUE THREAD

**Issue by jimmykane** [amcharts/amcharts4#966]
```
TypeError: undefined is not an object (evaluating 'points[count - 1].x')
```

```
TypeError: undefined is not an object (evaluating 'points[count - 1].x')
closeSegment — LineSeries.js:360
openSegment — LineSeries.js:313
validate — LineSeries.js:193
(anonymous function) — ValueAxis.js:823
each — Array.js:74
each — List.js:723
(anonymous function) — ValueAxis.js:822
(anonymous function) — EventDispatcher.js:266
each — Array.js:74
_eachListener — EventDispatcher.js:239
dispatchImmediately — EventDispatcher.js:264
end — Animation.js:495
update — Animation.js:604
(anonymous function) — System.js:282
each — Array.js:74
update — System.js:281
(anonymous function) — System.js:343
wrapped — raven.js:376
onInvokeTask — core.js:17279
runTask — zone.js:195
invokeTask — zone.js:498
timer — zone.js:2281
```

Here 
```
 if (baseAxis == xAxis) {
                closePoints.push({ x: points[count - 1].x, y: yAxis.basePoint.y }); // last x
                closePoints.push({ x: points[0].x, y: yAxis.basePoint.y }); // first x
            }
```

I am investigating this as well as we went live some days ago and some of these come up. 

That said the only reason I am opening an issue is to attract others, hear tipes and collab. 
Please dont be alerted

**Comment by jimmykane** [comment]
So I debugged the issue I am not sure what the ` LineSeries.prototype.closeSegment` function does but it has some issues in the code: 

This is the function: 

```
 if (baseAxis == xAxis) {
                closePoints.push({x: points[count - 1].x, y: yAxis.basePoint.y}); // last x
                closePoints.push({x: points[0].x, y: yAxis.basePoint.y}); // first x
              } else {
                closePoints.push({x: xAxis.basePoint.x, y: points[count - 1].y}); // last y
                closePoints.push({x: xAxis.basePoint.x, y: points[0].y}); // first y
              }
```
(I added a try catch to follow state) 


If the points are 0 then it breaks its so simple. 

In this part of code 
```
 var baseAxis = this.baseAxis;
              var count = points.length;
              var xAxis = this.xAxis;
              var yAxis = this.yAxis;
              if (baseAxis == xAxis) {
                closePoints.push({x: points[count - 1].x, y: yAxis.basePoint.y}); // last x
                closePoints.push({x: points[0].x, y: yAxis.basePoint.y}); // first x
              } else {
                closePoints.push({x: xAxis.basePoint.x, y: points[count - 1].y}); // last y
                closePoints.push({x: xAxis.basePoint.x, y: points[0].y}); // first y
              }
``` 

If `count` is `0` then `points[count - 1].x` fails

**Comment by jimmykane** [comment]
The same happens with 4.1.6 :-(

**Comment by martynasma** [comment]
How about with 4.1.8?

**Comment by jimmykane** [comment]
Will try it after my intervals. :-O 

Thanks for the update !

**Comment by jimmykane** [comment]
Nope. :-( That part of the code looks the same so the outer caller logically still breaks it .

**Comment by jimmykane** [comment]
And here is the stack that fires the error: 

```

zone.js:199 Uncaught TypeError: Cannot read property 'x' of undefined
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.closeSegment (LineSeries.js:375)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.openSegment (LineSeries.js:328)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.openSegmentWrapper (LineSeries.js:266)
    at LineSeries.push../node_modules/@amcharts/amcharts4/.internal/charts/series/LineSeries.js.LineSeries.validate (LineSeries.js:193)
    at ValueAxis.js:823
    at Module.each (Array.js:74)
    at List.push../node_modules/@amcharts/amcharts4/.internal/core/utils/List.js.List.each (List.js:723)
    at ValueAxis.js:822
    at Object.dispatch (EventDispatcher.js:381)
    at EventDispatcher.js:266
    at Module.each (Array.js:74)
    at EventDispatcher.push../node_modules/@amcharts/amcharts4/.internal/core/utils/EventDispatcher.js.EventDispatcher._eachListener (EventDispatcher.js:239)
    at EventDispatcher.push../node_modules/@amcharts/amcharts4/.internal/core/utils/EventDispatcher.js.EventDispatcher.dispatchImmediately (EventDispatcher.js:264)
    at Animation.push../node_modules/@amcharts/amcharts4/.internal/core/utils/Animation.js.Animation.end (Animation.js:495)
    at Animation.push../node_modules/@amcharts/amcharts4/.internal/core/utils/Animation.js.Animation.update (Animation.js:604)
    at System.js:282
    at Module.each (Array.js:74)
    at System.push../node_modules/@amcharts/amcharts4/.internal/core/System.js.System.update (System.js:281)
    at System.js:343
    at wrapped (raven.js:376)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:423)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.runTask (zone.js:195)
    at push../node_modules/zone.js/dist/zone.js.ZoneTask.invokeTask (zone.js:498)
    at ZoneTask.invoke (zone.js:487)
    at timer (zone.js:2281)
```

**Comment by jimmykane** [comment]
Additional info. 
This happens according to the following logic (also this logic can be causing the flickering in bug #960) 
- Add series with shared yAxis about 30 +
- After all the series are added, since they are many they are validated. 
- Then for all those series add data to those  

So adding data to each series after all series are added produces this error.

**Comment by jimmykane** [comment]
Ok both this issue and #960 are due to 
`series.simplifiedProcessing = true;`

**Comment by jimmykane** [comment]
Nope that was not it :-(

**Comment by jimmykane** [comment]
This is by far the most rising error I get on Sentry. It happens quite often and did not happen with the early 4.x.x versions.

**Comment by jimmykane** [comment]
Note, when this happens the whole charts crash. It also can crash silently. 

Looks like the part that determines the zoom level is causing this

**Comment by martynasma** [comment]
Fixed in [4.1.9](https://github.com/amcharts/amcharts4/releases/tag/4.1.9).

## [4.1.9] - 2019-02-20

### Added
- New `Sprite` property: `baseSprite`. On objects, even deep in hierarchy it will contain a reference to the main chart object.

### Changed
- Axis `min`/`max` calculation algorightm updated.

### Fixed
- On a chart with `DateAxis` and multiple series with same-date data items, bullets could sometimes disappear while scrolling the chart.
- On `DateAxis` with yearly granularity could sometimes show wrong year.
- Updating chart `data` with less data points was sometimes resulting in JavaScript error.
- `DateAxis` labels/ticks/grid was not placed in the correct position if `baseInterval.count > 1` and `location > 0`.
- Dynamically changing `ZoomControl` value of `layout` was not working correctly.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by jimmykane** [comment]
🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇 🥇  

Thank you!
